Sprint-3/implement/rotate-char.js

```diff
@@ -41,3 +41,35 @@ console.log(rotateCharacter("7", 5)); // Output: "7" (unchanged, not a letter)
 // And the function should return the rotated character as a string (e.g., 'z' rotated by 3 should become 'c', 'Z' rotated by 3 should become 'C').
 console.log(rotateCharacter("z", 1)); // Output: "a" (preserves case, but wraps around)
 console.log(rotateCharacter("Y", 2)); // Output: "A" (preserves case, but wraps around)
+
+
+function rotateCharacter(char, shift) {
+    // Check if the character is a lowercase letter
+    if (char >= 'a' && char <= 'z') {
+        let start = 'a'.charCodeAt(0); // ASCII code for 'a'
+        let offset = (char.charCodeAt(0) - start + shift) % 26;
+        return String.fromCharCode(start + (offset >= 0 ? offset : offset + 26));
```

Comment:
could you explain why is the offset>=0? check required? what test cases would fail without it?

Reply:
thak you for reviewing ,Without the offset >= 0 check, rotating a character with a negative shift would give a negative offset.

By adding offset >= 0 ? offset : offset + 26, we ensure that the offset always falls within the valid range of 0 to 25

Reply:
got it. The specs mention negative shifts only for non letter input, but this is solid as well.